shoulder/gadget/include_guard.py

```diff
@@ -20,22 +20,28 @@
 # OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 # SOFTWARE.
 
-from shoulder.gadget.abstract_gadget import AbstractGadget
-from shoulder.logger import logger
+def include_guard(_decorated=None, *, name="SHOULDER_AARCH64_H"):
+    """
+    A decorator gadget that generates include guards for C/C++ header files
 
-class IncludeGuardCloseGadget(AbstractGadget):
-    @property
-    def description(self):
-        return "Generate an closing include-guard for a C/C++ header file"
+    Args:
+        name (str): The name of the include guard to generate
 
-    def generate(self, objects, outfile):
-        outfile.write("#endif\n\n")
+    Usage:
+        @include_guard(name=MY_INCLUDE_GUARD_H)
```

Comment:
Does this default to "SHOULDER_AARCH64_H" if no name is provided (`@include_guard`)? I assume that the defaults are what is defined in the function definition above.

Reply:
Exactly, it will default to SHOULDER_AARCH64_H, and you can choose not to override the default by not specifying any arguments.